src/cpuinfo/SDL_cpuinfo.c

```diff
@@ -840,18 +844,40 @@ static const char *SDL_GetCPUName(void)
 int SDL_GetCPUCacheLineSize(void)
 {
     const char *cpuType = SDL_GetCPUType();
+    int cacheline_size;
     int a, b, c, d;
     (void)a;
     (void)b;
     (void)c;
     (void)d;
     if (SDL_strcmp(cpuType, "GenuineIntel") == 0 || SDL_strcmp(cpuType, "CentaurHauls") == 0 || SDL_strcmp(cpuType, "  Shanghai  ") == 0) {
         cpuid(0x00000001, a, b, c, d);
-        return ((b >> 8) & 0xff) * 8;
+	cacheline_size = ((b >> 8) & 0xff) * 8;
+        return cacheline_size;
     } else if (SDL_strcmp(cpuType, "AuthenticAMD") == 0 || SDL_strcmp(cpuType, "HygonGenuine") == 0) {
         cpuid(0x80000005, a, b, c, d);
-        return c & 0xff;
+	cacheline_size = c & 0xff;
+        return cacheline_size;
     } else {
+#if defined(__LINUX__) && defined (__GNUC__)
```

Comment:
I think `__GNUC__` is the wrong check here: You want to check `__GLIBC__`? (I's defined in features.h which is included by glibc headers included by SDL_cpuinfo.h)

Reply:
Oh thanks, that's correct I meant to check for glibc.